test/hpr_http_roaming_protocol_SUITE.erl

```diff
@@ -0,0 +1,318 @@
+            <<"DLMetaData">> => #{
+                <<"DevEUI">> => <<"0x6081f9c306a777fd">>,
+
+                <<"RXDelay1">> => RXDelay,
+                <<"DLFreq1">> => 925.1,
+                <<"DataRate1">> => 10,
+
+                <<"FNSULToken">> => Token,
+
+                <<"ClassMode">> => <<"A">>,
+                <<"HiPriorityFlag">> => false
+            }
+        }
+    end,
+
+    lists:foreach(
+        fun({RXDelay, ExpectedTimestamp}) ->
+            Input = MakeInput(RXDelay),
+            {downlink, _, {_, DownlinkPacket}, _} = hpr_http_roaming:handle_xmitdata_req(Input),
+            Timestamp = DownlinkPacket#packet_router_packet_down_v1_pb.rx1#window_v1_pb.timestamp,
+            ?assertEqual(ExpectedTimestamp, Timestamp)
+        end,
+        [
+            {0, 1_000_000},
+            {1, 1_000_000},
+            {2, 2_000_000},
+            {3, 3_000_000}
+        ]
+    ),
+
+    ok.
+
+rx1_downlink_test(_Config) ->
+    TransactionID = 17,
+    ok = hpr_http_roaming_utils:insert_handler(TransactionID, self()),
+
+    Payload = <<"0x60c04e26e020000000a754ba934840c3bc120989b532ee4613e06e3dd5d95d9d1ceb9e20b1f2">>,
+    RXDelay = 2,
+    FrequencyMhz = 925.1,
+    DataRate = 10,
+
+    Token = hpr_http_roaming:make_uplink_token(
+        TransactionID,
+        'US915',
+        erlang:system_time(millisecond),
+        <<"www.example.com">>,
+        sync
+    ),
+
+    Input = #{
+        <<"ProtocolVersion">> => <<"1.1">>,
+        <<"SenderID">> => <<"0x600013">>,
+        <<"ReceiverID">> => <<"0xc00053">>,
+        <<"TransactionID">> => rand:uniform(16#FFFF_FFFF),
+        <<"MessageType">> => <<"XmitDataReq">>,
+        <<"PHYPayload">> => Payload,
+        <<"DLMetaData">> => #{
+            <<"DevEUI">> => <<"0x6081f9c306a777fd">>,
+
+            <<"RXDelay1">> => RXDelay,
+            <<"DLFreq1">> => FrequencyMhz,
+            <<"DataRate1">> => DataRate,
+
+            <<"FNSULToken">> => Token,
+            <<"ClassMode">> => <<"A">>,
+            <<"HiPriorityFlag">> => false
+        }
+    },
+
+    {downlink, _Output, {Pid, DownlinkPacket}, _Dest} = hpr_http_roaming:handle_xmitdata_req(
+        Input
+    ),
+    ?assertEqual(Pid, self()),
+
+    PayloadFromDownlinkPacket = DownlinkPacket#packet_router_packet_down_v1_pb.payload,
+    ?assertEqual(
+        hpr_http_roaming_utils:hexstring_to_binary(Payload),
+        PayloadFromDownlinkPacket
+    ),
+    FrequencyFromDownlinkPacket =
+        DownlinkPacket#packet_router_packet_down_v1_pb.rx1#window_v1_pb.frequency,
+    ?assertEqual(true, (925_100_000 == FrequencyFromDownlinkPacket)),
+
+    %% signal strength not available
+    %%  ?assertEqual(27, blockchain_helium_packet_v1:signal_strength(Downlink)),
+
+    DatarateFromDownlinkPacket =
+        DownlinkPacket#packet_router_packet_down_v1_pb.rx1#window_v1_pb.datarate,
+    ?assertEqual(
+        hpr_lorawan:index_to_datarate('US915', DataRate),
+        DatarateFromDownlinkPacket
+    ),
```

Comment:
Use the accessors in `hpr_packet_down`.

Responsibility for RSSI has been moved to the gateway which is region aware, this comment can be removed.

```
?assertEqual(true, (925_100_000 == FrequencyFromDownlinkPacket)),
```
If we write it as
```
?assertEqual(925_100_000, FrequencyFromDownlinkPacket),
```
we have a better chance of gleaning something from a failure without having to come in and add logging.
Or getting a sense if it might be a false-positive that needs to be addressed.

Reply:
```Failure/Error: ?assertEqual(925100000, FrequencyFromDownlinkPacket)
  expected: 925100000
       got: 9.251e8
      line: 251```
      
The only way around this that I could think of was to use '=='

Reply:
"Use the accessors in hpr_packet_down.

Responsibility for RSSI has been moved to the gateway which is region aware, this comment can be removed."

--Done.

Reply:
Used assertMatch.